### ISSUE TITLE
tmux nerdfont rendering within Neovim

### ISSUE THREAD

**Issue by cheeze2000** [tmux/tmux#3344]
### Issue description

https://user-images.githubusercontent.com/26966445/194215460-c7b2963a-aff3-467e-b167-de0568516083.mp4

Within tmux, there are problems rendering nerdfont icons in Neovim as shown in the clip. Outside of tmux, the icons are rendered perfectly fine.

### Required information

Please provide the following information:

* tmux version (`tmux -V`). 
tmux 3.3a
* Platform (`uname -sp`).
Linux x86_64
* $TERM inside and outside of tmux (`echo $TERM`).
Outside: xterm-256color
Inside: tmux-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`).
[tmux-client-7070.log](https://github.com/tmux/tmux/files/9721270/tmux-client-7070.log)
[tmux-out-7072.log](https://github.com/tmux/tmux/files/9721271/tmux-out-7072.log)
[tmux-server-7072.log](https://github.com/tmux/tmux/files/9721272/tmux-server-7072.log)

**Comment by nicm** [comment]
Try building tmux with utf8proc.

**Comment by cheeze2000** [comment]
> Try building tmux with utf8proc.

`./configure --enable-utf8proc && make`
`sudo make install`

Sorry, no luck. The problem still persists.

**Comment by nicm** [comment]
If you put one of the characters in `status-left` does it render correctly or is the status line too short or too long?

**Comment by cheeze2000** [comment]
> If you put one of the characters in `status-left` does it render correctly or is the status line too short or too long?

For the video above, I did not use my config file. I have a .tmux.conf with those nerdfont icons and it renders correctly.

**Comment by nicm** [comment]
It seems likely vim has a different idea about the width of these characters than tmux. If the status line is showing correctly, then tmux must know the correct width.

There are dozens of double width characters in this log so I can't tell what is right and what is wrong. Can you show me a log from printing one of these with `printf` (eg `printf xCCx` where CC is the bad character)?

**Comment by cheeze2000** [comment]
> It seems likely vim has a different idea about the width of these characters than tmux. If the status line is showing correctly, then tmux must know the correct width.
> 
> There are dozens of double width characters in this log so I can't tell what is right and what is wrong. Can you show me a log from printing one of these with `printf` (eg `printf xCCx` where CC is the bad character)?
[tmux-server-11074.log](https://github.com/tmux/tmux/files/9721456/tmux-server-11074.log)
[tmux-client-11072.log](https://github.com/tmux/tmux/files/9721457/tmux-client-11072.log)
[tmux-out-11074.log](https://github.com/tmux/tmux/files/9721458/tmux-out-11074.log)

`tmux -vv new`
`printf xx`
`exit`

I have a feeling this is something to do with Neovim.

**Comment by cheeze2000** [comment]
![image](https://user-images.githubusercontent.com/26966445/194222557-f43e7a83-62cd-4673-9c11-b5572397bb8d.png)
![image](https://user-images.githubusercontent.com/26966445/194222662-cb979298-b88d-4ff2-9848-0ec5093f1482.png)
Seems like width 1 to me. On another note, I seem to experience the same issue as this https://github.com/neovim/neovim/issues/14760

**Comment by cheeze2000** [comment]
```
x✨x
x☁️x
```

(I changed my font to a non-nerdfont version to make sure these are not nerdfont icons)
I have narrowed down the problem to these two characters, they have different widths and they are not even Nerdfont icons.
![image](https://user-images.githubusercontent.com/26966445/194229772-b5b4c38e-e506-4bb1-8739-ca2368b36605.png)

There is some screen tearing on the line number, I am going to assume tmux (or utf8proc) thinks the cloud is width 2?
https://user-images.githubusercontent.com/26966445/194230028-0597b503-429b-4bb3-8859-0050f28cd62a.mp4

Feel free to close this issue, it seems like only Neovim can fix this.
